File: test/benchmark/larceny/diviter.js
// Larceny Benchmark: diviter (Node.js)
// Iterative integer division via repeated subtraction
'use strict';

function diviterDiv(x, y) {
    let q = 0;
    let r = x;
    while (r >= y) {
        r -= y;
        q++;
    }
    return q;
}

function diviterMod(x, y) {
    let r = x;
    while (r >= y) {
        r -= y;
    }
    return r;
}

function main() {
    const __t0 = performance.now();
    let result = 0;
    for (let iter = 0; iter < 1000; iter++) {
        result += diviterDiv(1000000, 2);
        result -= diviterMod(1000000, 2);
    }
    const __t1 = performance.now();
    if (result === 500000000) {
        process.stdout.write("diviter: PASS\n");
    } else {
        process.stdout.write("diviter: FAIL result=" + result + "\n");
    }
    process.stdout.write("__TIMING__:" + (__t1 - __t0) + "\n");
}

main();

File: test/benchmark/larceny/python/diviter.py
#!/usr/bin/env python3
# Larceny Benchmark: diviter (Python)
# Iterative integer division via repeated subtraction
import time


def diviter_div(x, y):
    q = 0
    r = x
    while r >= y:
        r -= y
        q += 1
    return q


def diviter_mod(x, y):
    r = x
    while r >= y:
        r -= y
    return r


def main():
    t0 = time.perf_counter_ns()
    result = 0
    for _ in range(1000):
        result += diviter_div(1000000, 2)
        result -= diviter_mod(1000000, 2)
    t1 = time.perf_counter_ns()

    if result == 500000000:
        print("diviter: PASS")
    else:
        print(f"diviter: FAIL result={result}")
    print(f"__TIMING__:{(t1 - t0) / 1e6:.3f}")


main()
